Expo CLI (0.3.2) depends on incompatible version of @expo/dev-server (0.1.122)

### Summary

There is a weird issue of module dependency when starting by "expo start":
Error: Cannot find module '@expo/dev-server/build/webpack/symbolicateMiddleware'

Normal case that modules installed via Yarn 1.x

```
└─┬ expo@46.0.16
  └─┬ @expo/cli@0.3.2
    └── @expo/dev-server@0.1.120
```

After upgrade to Yarn 3.x and re-install all modules.

```
└─┬ expo@46.0.16
  └─┬ @expo/cli@0.3.2
    └── @expo/dev-server@0.1.122
```

My workaround is to install specified version (downgrade @expo/dev-server to 0.1.120)
<img width="291" alt="image" src="https://user-images.githubusercontent.com/7246502/199709680-71c7ee72-668c-4bfe-8f1a-ed1f9f87fc4b.png">


### What platform(s) does this occur on?

_No response_

### SDK Version

46

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 11.6.6
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 16.17.1 - ~/.nvm/versions/node/v16.17.1/bin/node
      Yarn: 3.2.4 - ~/.nvm/versions/node/v16.17.1/bin/yarn
      npm: 8.15.0 - ~/.nvm/versions/node/v16.17.1/bin/npm
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    IDEs:
      Android Studio: 4.1 AI-201.8743.12.41.7199119
      Xcode: 12.5/12E262 - /usr/bin/xcodebuild
    npmPackages:
      @expo/webpack-config: ^0.17.0 => 0.17.3
      expo: ~46.0.16 => 46.0.16
      react: 18.0.0 => 18.0.0
      react-dom: 18.0.0 => 18.0.0
      react-native: 0.69.6 => 0.69.6
      react-native-web: ~0.18.7 => 0.18.10
    Expo Workflow: managed
```

### Minimal reproducible example

Just upgrade Yarn from 1.x to 3.x.
https://yarnpkg.com/getting-started/install#initializing-your-project

```sh
yarn init -2
```

<img width="250" alt="image" src="https://user-images.githubusercontent.com/7246502/199708759-bf1622f1-384f-4a3b-a4e1-84aee4008226.png">

<img width="284" alt="image" src="https://user-images.githubusercontent.com/7246502/199708967-7cdf57d3-8272-41a4-bc90-f7d05cbe1b3a.png">

--- a/packages/@expo/cli/CHANGELOG.md
+++ b/packages/@expo/cli/CHANGELOG.md
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Pin `@expo/dev-server` to version `0.1.120` to keep webpack support in SDK 46 ([#20878](https://github.com/expo/expo/pull/20878) by [@byCedric](https://github.com/byCedric))
+
 ### 💡 Others
 
 ## 0.3.2 — 2022-10-13

--- a/packages/@expo/cli/package.json
+++ b/packages/@expo/cli/package.json
@@ -40,7 +40,7 @@
     "@expo/code-signing-certificates": "^0.0.2",
     "@expo/config": "~7.0.1",
     "@expo/config-plugins": "~5.0.1",
-    "@expo/dev-server": "~0.1.120",
+    "@expo/dev-server": "0.1.120",
     "@expo/devcert": "^1.0.0",
     "@expo/json-file": "^8.2.35",
     "@expo/metro-config": "~0.4.0",

